lib/SQL/Abstract/mysql.pm

```diff
@@ -98,6 +140,95 @@ sub _order_by {
   return $sql, @bind;
 }
 
+sub _order_by_chunks {
+  my ($self, $arg) = @_;
+  if ($arg =~ /->/) {
+    return $self->_json_extract($arg);
+  }
+  else {
+    return $self->SUPER::_order_by_chunks($arg);
+  }
+}
+
+sub _select_fields {
+  my ($self, $fields) = @_;
+  return $fields unless ref $fields eq 'ARRAY';
+
+  if (grep /->/, @$fields) {
+    return join ',', map { $self->_json_extract($_, 1) } @$fields;
+  }
+  else {
+    return join ',', map { $self->_quote($_) } @$fields;
+  }
+}
+
+sub _update_set_values {
```

Comment:
I think this whole method is a bit hard to read, because of all the nested if/else. Can you move some of the code out in methods with descriptive names? I think that could limit the nesting by using `return` instead.

Reply:
It's hard to read, but there are too many variables for a method and the block is only 40 lines long. Maybe some comments will help?